src/coreComponents/dataRepository/LogLevelsInfo.hpp

```diff
@@ -0,0 +1,193 @@
+ */
+
+/**
+ * @file LogLevelsInfo.hpp
+ * This file contains all log level informations and the mecanism to ensure LOG_LEVEL_INFO structure is valid
+ */
+#ifndef GEOS_COMMON_LOGLEVELSINFO_HPP
+#define GEOS_COMMON_LOGLEVELSINFO_HPP
+
+#include "common/DataTypes.hpp"
+#include "common/format/Format.hpp"
+
+namespace geos
+{
+
+namespace logInfo
+{
+
+/**
+ * @name Common LogLevels info structures. They must comply with the `is_log_level_info` trait.
+ */
+///@{
+
+/// @cond DO_NOT_DOCUMENT
+struct LineSearch
+{
+  static constexpr int getMinLogLevel() { return 1; }
+  static constexpr std::string_view getDescription() { return "Information on line search"; }
+};
+
+struct LineSearchFailed
+{
+  static constexpr int getMinLogLevel() { return 1; }
+  static constexpr std::string_view getDescription() { return "On Incorrect solution, Information on failed line search"; }
+};
+
+struct ScalingFactor
+{
+  static constexpr int getMinLogLevel() { return 1; }
+  static constexpr std::string_view getDescription() { return "Information on global solution scaling factor"; }
+};
+
+struct TimeStep
+{
+  static constexpr int getMinLogLevel() { return 1; }
+  static constexpr std::string_view getDescription() { return "Information on the timestep"; }
+};
+
+struct SolverTimers
+{
+  static constexpr int getMinLogLevel() { return 1; }
+  static constexpr std::string_view getDescription() { return "Information on solver timers"; }
+};
+
+struct ScreenLinearSystem
+{
+  static constexpr int getMinLogLevel() { return 1; }
+  static constexpr std::string_view getDescription() { return "Output to screen the assembled linear system and solutions (matrices and vectors)"; }
+
+};
+
+struct FileLinearSystem
+{
+  static constexpr int getMinLogLevel() { return 2; }
+  static constexpr std::string_view getDescription() { return "Output to file the assembled linear system and solutions (matrices and vectors)"; }
+};
+
+struct SolverConfig
+{
+  static constexpr int getMinLogLevel() { return 1; }
+  static constexpr std::string_view getDescription() { return "On non convergance, Information about testing new configuration and print the time step"; }
+};
+
+struct ResidualNorm
+{
+  static constexpr int getMinLogLevel() { return 1; }
+  static constexpr std::string_view getDescription() { return "Print residual norm"; }
+};
+
+struct ResidualValues
+{
+  static constexpr int getMinLogLevel() { return 1; }
+  static constexpr std::string_view getDescription() { return "Print the residual values"; }
+};
+
+struct LinearSystem
+{
+  static constexpr int getMinLogLevel() { return 1; }
+  static constexpr std::string_view getDescription() { return "Information oon linear system"; }
+};
+
+struct CrossflowWarning
+{
+  static constexpr int getMinLogLevel() { return 1; }
+  static constexpr std::string_view getDescription() { return "If the well is injector and crossflow enabled, display informations about crossflow for injectors"; }
```

Comment:
I don't understand the choice of what is being defined as struct in this file and what is not - there are more general categories like line search or linear solver, but then this crossflow warning - very specific, why?

Reply:
The aim is to have an overview of what loglevels bring us, which is why I've remained vague about LineSearch, but the description can be rephrase